src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1952,9 +1959,9 @@ private void ExecuteCompilationActions(
             }
         }
 
-        internal static Action<Diagnostic, CancellationToken> GetDiagnosticSink(Action<Diagnostic> addDiagnosticCore, Compilation compilation, AnalyzerOptions? analyzerOptions, SeverityFilter severityFilter, ConcurrentSet<string>? suppressedDiagnosticIds)
+        internal static Action<Diagnostic, AnalyzerOptions, CancellationToken> GetDiagnosticSink(Action<Diagnostic> addDiagnosticCore, Compilation compilation, SeverityFilter severityFilter, ConcurrentSet<string>? suppressedDiagnosticIds)
         {
-            return (diagnostic, cancellationToken) =>
+            return (diagnostic, analyzerOptions, cancellationToken) =>
```

Comment:
these are all now passed in options dynamically in the callback, isntead of capturing the global ones.